### PR TITLE
[ggj]feat: while statement implementation

### PR DESCRIPTION
**Pull request by xiaozhenliu-gg5**

### CODE REVIEW

File: src/main/java/com/google/api/generator/engine/writer/JavaWriterVisitor.java [hunk 1]
```diff
@@ -162,6 +164,19 @@ public void visit(ExprStatement exprStatement) {
...
+  @Override
+  public void visit(WhileStatement whileStatement) {
+    buffer.append("while (");
+    whileStatement.conditionExpr().accept(this);
+    buffer.append(") {");
+    newline();
+    for (Statement statement : whileStatement.body()) {
+      statement.accept(this);
+    }
+    buffer.append("} ");
```
Comment (miraleung) [inline src/main/java/com/google/api/generator/engine/writer/JavaWriterVisitor.java:176]: Nit: Why an extra space after `}`?

Comment (xiaozhenliu-gg5) [reply src/main/java/com/google/api/generator/engine/writer/JavaWriterVisitor.java:176]: done. Also cleaned up other `} `, thanks!

File: src/main/java/com/google/api/generator/engine/ast/WhileStatement.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+    public abstract Builder setBody(List<Statement> body);
+
+    abstract WhileStatement autoBuild();
+
+    public WhileStatement build() {
+      WhileStatement whileStatement = autoBuild();
+      Preconditions.checkState(
+          whileStatement.conditionExpr().type().equals(TypeNode.BOOLEAN),
+          "While-condition must be a boolean-typed expression");
```
Comment (miraleung) [inline src/main/java/com/google/api/generator/engine/ast/WhileStatement.java:45]: Nit: "While condition"

Comment (xiaozhenliu-gg5) [reply src/main/java/com/google/api/generator/engine/ast/WhileStatement.java]: done. Thanks!

**Pull request merged**